### PR TITLE
Add unit tests for functions within `.~/data/utils.js`

### PR DESCRIPTION
**Pull request by eason9487**
### Changes proposed in this Pull Request:

Add unit tests for functions within `.~/data/utils.js`

- Add test cases for the returning `ids` field of `getReportQuery` function
- Add test cases for `getReportKey`, `calculateDelta`, and `mapReportFieldsToPerformance` functions

### Detailed test instructions:

1. Run `npm run test-unit` and make sure all these tests pass

### CODE REVIEW

File: js/src/data/utils.test.js [hunk 1]
```diff
@@ -65,4 +73,180 @@ describe( 'getReportQuery', () => {
...
 		} );
 	} );
+
+	it( 'When the query does not contain `programs` or `products`, should have no `ids` field', () => {
+		forAnyCategoryAndDateReference( ( category, dateReference ) => {
+			const query = {};
+
+			expect(
+				getReportQuery( category, 'free', query, dateReference ).ids
+			).toBeUndefined();
```
Comment (tomalec) [inline js/src/data/utils.test.js]: :nail_care: If we are specing "should have no `ids` field", I'd rather use `not.toHaveProperty`
```suggestion
			expect(
				getReportQuery( category, 'free', query, dateReference )
			).not.toHaveProperty( 'id' );
```

Comment (tomalec) [reply js/src/data/utils.test.js]: Same applies for the ones below, `like `.toHaveProperty( 'ids', query.programs )` - it gives more meaningful errors when fail

File: js/src/data/utils.test.js [hunk 2]
```diff
@@ -65,4 +73,180 @@ describe( 'getReportQuery', () => {
...
+		const query = {
+			programs: '123,456',
+		};
+
+		expect(
+			getReportQuery( 'programs', 'free', query, 'primary' ).ids
+		).toBe( query.programs );
+	} );
+
+	it( 'When the products report query contains `products`, should have transformed `ids` field', () => {
```
Comment (tomalec) [inline js/src/data/utils.test.js]: :nail_care: I'd explain what "transformation" is expected.
```suggestion
	it( 'When the products report query contains `products`, should have `ids` field, with values prepended with `gla_`', () => {
```

**Comment by tomalec** [comment]
:scroll: :nail_care: If we aim for fuller coverage, the default values of `mapReportFieldsToPerformance` are not covered.

```
npm run test-unit -- js/src/data/utils.test.js --coverage

-----------|---------|----------|---------|---------|-------------------
File       | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s 
-----------|---------|----------|---------|---------|-------------------
All files  |     100 |    93.75 |     100 |     100 |                   
 data      |     100 |    93.55 |     100 |     100 |                   
  utils.js |     100 |    93.55 |     100 |     100 | 111-112           
 utils     |     100 |      100 |     100 |     100 |                   
  round.js |     100 |      100 |     100 |     100 |                   
-----------|---------|----------|---------|---------|-------------------

```

File: js/src/data/utils.test.js (hunk 1 above)
Comment (eason9487) [reply js/src/data/utils.test.js]: > If we are specing "should have no `ids` field", I'd rather use `not.toHaveProperty`

Agreed, that's the correct way to verify this case.

> it gives more meaningful errors when fail

Thanks! Updated by https://github.com/woocommerce/google-listings-and-ads/pull/856/commits/9e3533f5a3b3c9f82e7ba7e67a9ad9ac1f7fa86a.

**Comment by eason9487** [comment]
> If we aim for fuller coverage, the default values of `mapReportFieldsToPerformance` are not covered.

Added by https://github.com/woocommerce/google-listings-and-ads/pull/856/commits/836796374d9684ff351cde6192017605ff9ad1ea

**Review by tomalec — APPROVED** [review]
LGTM, thanks :)

**Pull request merged**